pkg/client/observatorium/observability_config.go

```diff
@@ -94,6 +135,52 @@ func (c *ObservabilityCloudWatchLoggingConfig) validate() error {
 	return nil
 }
 
+func validateExactlyOneOIDCTypeProvided(sl validator.StructLevel) {
+	config := sl.Current().Interface().(DataPlaneObservabilityRemoteWriteAuthTypeOIDC)
+	if config.Static == nil && config.Autogenerated == nil {
+		sl.ReportError(config, "static_configuration", "Static", "one_of", "")
+		sl.ReportError(config, "autogenerated_configuration", "Autogenerated", "one_of", "")
+	}
+
+	if config.Static != nil && config.Autogenerated != nil {
+		sl.ReportError(config, "static_configuration", "Static", "only_one", "")
+		sl.ReportError(config, "autogenerated_configuration", "Autogenerated", "only_one", "")
+	}
+}
+
+func (c *DataPlaneObservabilityConfig) validate() error {
+	if !c.Enabled {
+		return nil
+	}
+
+	validate := validator.New()
+	validate.RegisterStructValidation(validateExactlyOneOIDCTypeProvided, DataPlaneObservabilityRemoteWriteAuthTypeOIDC{})
```

Comment:
We also need to validate that exactly one Authentication type is provided

Reply:
Maybe the `required` validate struct tag already covers this. Could you verify it?

Reply:
> Isn't that already the case? We're failing the validation if both are nil or if both are not nil. That only allows cases where exactly one is defined. If we later add a third option, we will have to update the validator anyway.

You are describing the validation around the possible OIDC configurations. What I am describing is the validation around the possible Authentications